File: Clients/ios/DiveLog/ViewModels/EquipmentViewModel.swift
import Foundation

@MainActor
class EquipmentViewModel: ObservableObject {
    @Published var equipment: [EquipmentItem] = []
    @Published var isLoading = false
    @Published var errorMessage: String?

    func loadEquipment() async {
        isLoading = true
        errorMessage = nil
        do {
            equipment = try await APIService.shared.fetchEquipment()
        } catch {
            errorMessage = error.localizedDescription
        }
        isLoading = false
    }

    func deleteEquipment(_ item: EquipmentItem) async {
        do {
            try await APIService.shared.deleteEquipment(id: item.id)
            equipment.removeAll { $0.id == item.id }
        } catch {
            errorMessage = error.localizedDescription
        }
    }
}

File: Clients/android/app/src/main/java/com/divelog/app/viewmodel/EquipmentViewModel.kt
package com.divelog.app.viewmodel

import androidx.lifecycle.ViewModel
import androidx.lifecycle.viewModelScope
import com.divelog.app.data.api.ApiService
import com.divelog.app.data.model.EquipmentItem
import kotlinx.coroutines.flow.MutableStateFlow
import kotlinx.coroutines.flow.asStateFlow
import kotlinx.coroutines.launch

class EquipmentViewModel : ViewModel() {
    private val _equipment = MutableStateFlow<List<EquipmentItem>>(emptyList())
    val equipment = _equipment.asStateFlow()

    private val _isLoading = MutableStateFlow(false)
    val isLoading = _isLoading.asStateFlow()

    private val _error = MutableStateFlow<String?>(null)
    val error = _error.asStateFlow()

    fun loadEquipment() {
        viewModelScope.launch {
            _isLoading.value = true
            _error.value = null
            try {
                _equipment.value = ApiService.fetchEquipment()
            } catch (e: Exception) {
                _error.value = e.message
            }
            _isLoading.value = false
        }
    }

    fun deleteEquipment(item: EquipmentItem) {
        viewModelScope.launch {
            try {
                ApiService.deleteEquipment(item.id)
                _equipment.value = _equipment.value.filter { it.id != item.id }
            } catch (e: Exception) {
                _error.value = e.message
            }
        }
    }
}
